docs/architecture/adr-042-group-module.md

```diff
@@ -0,0 +1,375 @@
+
+```proto
+message GroupMember {
+
+    // group_id is the unique ID of the group.
+    uint64 group_id = 1;
+
+    // member is the member data.
+    Member member = 2;
+}
+
+// Member represents a group member with an account address,
+// non-zero weight and metadata.
+message Member {
+
+    // address is the member's account address.
+    string address = 1;
+    
+    // weight is the member's voting weight that should be greater than 0.
+    string weight = 2;
+    
+    // metadata is any arbitrary metadata to attached to the member.
+    bytes metadata = 3;
+}
+```
+
+### Group Account
+
+A group account is an account associated with a group and a decision policy.
+A group account does have a balance.
+
+Group accounts are abstracted from groups because a single group may have
+multiple decision policies for different types of actions. Managing group
+membership separately from decision policies results in the least overhead
+and keeps membership consistent across different policies. The pattern that
+is recommended is to have a single master group account for a given group,
+and then to create separate group accounts with different decision policies
+and delegate the desired permissions from the master account to
+those "sub-accounts" using the [`x/authz` module](adr-030-authz-module.md).
+
+Group accounts are stored as part of the `groupAccountTable` and modeled by `GroupAccountInfo`.
+
+```proto
+message GroupAccountInfo {
+
+    // address is the group account address.
+    string address = 1;
+    
+    // group_id is the unique ID of the group.
+    uint64 group_id = 2;
+
+    // admin is the account address of the group admin.
+    string admin = 3;
+    
+    // metadata is any arbitrary metadata to attached to the group account.
+    bytes metadata = 4;
+
+    // version is used to track changes to a group's GroupAccountInfo structure that
+    // would create a different result on a running proposal.
+    uint64 version = 5;
+
+    // decision_policy specifies the group account's decision policy.
+    google.protobuf.Any decision_policy = 6 [(cosmos_proto.accepts_interface) = "DecisionPolicy"];
+}
+```
+
+The group account address is generated based on an auto-increment integer which is used to derive the group module `RootModuleKey` into a `DerivedModuleKey`, as stated in [ADR-033](adr-033-protobuf-inter-module-comm.md#modulekeys-and-moduleids). The group account is added as a new `ModuleAccount` through `x/auth`.
+
+### Decision Policy
+
+A decision policy is the mechanism by which members of a group can vote on 
+proposals.
+
+All decision policies generally would have a minimum and maximum voting window.
+The minimum voting window is the minimum amount of time that must pass in order
+for a proposal to potentially pass, and it may be set to 0. The maximum voting
+window is the maximum time that a proposal may be voted on before it is closed.
+Both of these values must be less than a chain-wide max voting window parameter.
```

Comment:
```suggestion
Voting for each proposal is possible for at least the minimum voting window (can be 0) 
and not more than the maximum voting window. Proposal can be executed after the 
minimum voting window if it reached enough support. 
Both of these values must be less than a chain-wide max voting window parameter.
```

Reply:
I find the suggestion a bit confusing and prefer the original phrasing with 2 different sentences: one to explain the min voting window and another one for the max voting window. But I can add something about proposal execution:
```suggestion
The minimum voting window is the minimum amount of time that must pass in order
for a proposal to potentially pass, and it may be set to 0. The maximum voting
window is the maximum time that a proposal may be voted on and executed if 
it reached enough support before it is closed.
Both of these values must be less than a chain-wide max voting window parameter.
```

Reply:
Looks better. I would change _minimum amount of time_ to _minimum duration_

I don't like this kind of sentences (from original version): _The minimum voting window is the minimum amount of time that must pass in order for a proposal to potentially pass, and it may be set to 0._  - it's too much repetition and too complex to read.

Reply:
ok updated